packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -3107,3 +3121,20 @@ class _SelectableFragment with Selectable, Diagnosticable, ChangeNotifier implem
     properties.add(DiagnosticsProperty<String>('fullText', fullText));
   }
 }
+
+class _SelectableFragmentSelectedContentRange extends SelectedContentRange<String> {
+  _SelectableFragmentSelectedContentRange({
+    int start = -1,
+    int end = -1,
+    required super.content,
+  }) : _start = start,
+       _end = end;
+
+  @override
+  int get startOffset => _start;
+  final int _start;
+
+  @override
+  int get endOffset => _end;
+  final int _end;
+}
```

Comment:
This is private so the API doesn't matter much, but...

Why do you have to use separate `start` and `end` instead of `startOffset` and `endOffset` when they're pretty much the same?

Or even better, it seems like you could just about make SelectedContentRange non-abstract and use SelectedContentRange<String> and get rid of _SelectableFragmentSelectedContentRange? Or is there something preventing that?

Reply:
This subclass no longer exists.